docs/plugins/development.md

```diff
@@ -0,0 +1,243 @@
+)
+```
+
+Many of these are self-explanatory, but for more information, see the [setuptools documentation](https://setuptools.readthedocs.io/en/latest/setuptools.html).
+
+The key requirement for a NetBox plugin is the presence of an entry point for `netbox_plugins` pointing to the `PluginConfig` subclass, which we'll define next.
+
+### Define a PluginConfig
+
+The `PluginConfig` class is a NetBox-specific wrapper around Django's built-in [`AppConfig`](https://docs.djangoproject.com/en/stable/ref/applications/). It is used to declare NetBox plugin functionality within a Python package. Each plugin should provide its own subclass, defining its name, metadata, and default and required configuration parameters. An example is below:
+
+```python
+from extras.plugins import PluginConfig
+
+class AnimalSoundsConfig(PluginConfig):
+    name = 'netbox_animal_sounds'
+    verbose_name = 'Animal Sounds Plugin'
+    version = '0.1'
+    author = 'Author Name'
+    description = 'Show animals and the sounds they make'
+    url_slug = 'animal-sounds'
+    required_settings = []
+    default_settings = {
+        'loud': False
+    }
+```
+
+#### PluginConfig Attributes
+
+* `name` - Raw plugin name; same as the plugin's source directory
+* `author_name` - Name of plugin's author
+* `verbose_name` - Human-friendly name
+* `version` - Plugin version
+* `description` - Brief description of the plugin's purpose
+* `url_slug` - Base path to use for plugin URLs (optional). If not specified, the project's `name` will be used.
+* `required_settings`: A list of configuration parameters that **must** be defined by the user
+* `default_settings`: A dictionary of configuration parameter names and their default values
+* `min_version`: Minimum version of NetBox with which the plugin is compatible
+* `max_version`: Maximum version of NetBox with which the plugin is compatible
+* `middleware`: A list of middleware classes to append after NetBox's build-in middleware.
+* `caching_config`: Plugin-specific cache configuration
+
+## Database Models
+
+Plugins can define their own Django models to record user data. A model is a Python representation of a database table. Model instances can be created, manipulated, and deleted using the [Django ORM](https://docs.djangoproject.com/en/stable/topics/db/). Models are typically defined within a plugin's `models.py` file, though this is not a strict requirement.
+
+Below is a simple example `models.py` file showing a model with two character fields:
+
+```python
+from django.db import models
+
+class Animal(models.Model):
+    name = models.CharField(max_length=50)
+    sound = models.CharField(max_length=50)
+
+    def __str__(self):
+        return self.name
+```
+
+Once you have defined the model(s) for your plugin, you'll need to create the necessary database schema migrations as well. This can be done using the Django `makemigrations` management command:
+
+```no-highlight
+$ ./manage.py makemigrations netbox_animal_sounds 
```

Comment:
They will need to install the plugin package as it exists before this will work.